### PR TITLE
Modify the behaviour of S6323

### PR DESCRIPTION
**Pull request by stanislav-jordanov-sonarsource**
There was a misinterpretation of the description of SONARPHP-1274.
This resulted in https://github.com/SonarSource/rspec/pull/779/files and this PR

### CODE REVIEW

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java [hunk 1]
```diff
@@ -31,24 +35,66 @@
 
 public class EmptyAlternativeFinder extends RegexBaseVisitor {
 
-  private static final String MESSAGE = "Remove this empty alternative.";
+  private static final String MESSAGE_REMOVE_THIS = "Remove this empty alternative.";
```
Comment (nils-werner-sonarsource) [inline regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java:38]: Why did you change the constant name? MESSAGE is clear in almost all check aka finder contexts. Something which is pre or postfix with "REMOVE_THIS" looks more like a leftover or ugly TODO. I've got your idea, but I really don't like it.

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java [hunk 2]
```diff
@@ -31,24 +35,66 @@
...
+    super.visitAtomicGroup(tree);
+    nestedGroupLevel--;
+  }
+
+  @Override
+  public void visitLookAround(LookAroundTree tree) {
+    nestedGroupLevel++;
+    super.visitLookAround(tree);
+    nestedGroupLevel--;
+  }
```
Comment (nils-werner-sonarsource) [inline regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java:74]: It's the second time, that we have to visit all sub kinds of GroupTree maybe it's worth creating a visitGroup method in RegexBaseVisitor. WDYT? I would make both Finder implementations more readable.

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java [hunk 3]
```diff
@@ -31,24 +35,66 @@
...
-    for (int i = 0; i < alternatives.size() - 1; i++) {
+    int nAlternatives = alternatives.size();
+    boolean firstIsEmpty = false;
+    boolean lastIsEmpty = false;
+    for (int i = 0; i < nAlternatives; i++) {
       if (isEmptyAlternative(alternatives.get(i))) {
-        SourceCharacter orOperator = tree.getOrOperators().get(i);
-        regexElementIssueReporter.report(orOperator, MESSAGE, null, Collections.emptyList());
+        firstIsEmpty |= (i == 0);
+        lastIsEmpty |= (i == nAlternatives -1);
```
Comment (nils-werner-sonarsource) [inline regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java:85]:
```suggestion
        lastIsEmpty |= (i == nAlternatives - 1);
```

**Review by nils-werner-sonarsource — APPROVED** [review]
LGTM! Please see comments

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java (hunk 1 above)
Comment (stanislav-jordanov-sonarsource) [reply regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java:38]: To give you an idea how it got here:
That's a leftover from the brief time I had couple of different messages MSG_REMOVE_THIS & MSG_REMOVE_EITHER

File: regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java (hunk 2 above)
Comment (stanislav-jordanov-sonarsource) [reply regex-parsing/src/main/java/org/sonarsource/analyzer/commons/regex/finders/EmptyAlternativeFinder.java:74]: Yes, I had the same though.
